palantirnet/federated-search-react
# Security updates

## Notes

- `urijs` doesn't appear in `yarn.lock` but is flagged as a vulnerability in https://github.com/palantirnet/federated-search-react/security/dependabot/yarn.lock/urijs/open. The bot is likely finding the hash on this line as a false positive: https://github.com/palantirnet/federated-search-react/blob/d8259bfb95b31d34b48e65af3ede9bacd27e79e6/yarn.lock#L10111
- Updated to `react-scripts` version 4.0.1 to get
    - `node-notifier` 8.0.1 (https://github.com/palantirnet/federated-search-react/security/dependabot/yarn.lock/node-notifier/open)
    - `node-forge` 0.10.0 (https://github.com/palantirnet/federated-search-react/security/dependabot/yarn.lock/node-forge/open)
    - `yargs-parser` 13.1.2 (https://github.com/palantirnet/federated-search-react/security/dependabot/yarn.lock/yargs-parser/open)
    - `serialize-javascript` 4.0.0 (https://github.com/palantirnet/federated-search-react/security/dependabot/yarn.lock/serialize-javascript/open)
    - `lodash` 4.17.20 (https://github.com/palantirnet/federated-search-react/security/dependabot/yarn.lock/lodash/open)
- Updated to `patch-package` 6.2.2 to get `dot-prop` 5.3.0 (https://github.com/palantirnet/federated-search-react/security/dependabot/yarn.lock/dot-prop/open)

## Testing

- `yarn install`
- Setup https://github.com/palantirnet/federated-search-demo as directed in the readme there.
- Enable CORS for the D8 demo site by adding the following to `services.yml`:
```
  cors.config:
    enabled: true
    # Specify allowed headers, like 'x-allowed-header'.
    allowedHeaders: ['*']
    # Specify allowed request methods, specify ['*'] to allow all possible ones.
    allowedMethods: ['*']
    # Configure requests allowed from specific origins.
    allowedOrigins: ['*']
    # Sets the Access-Control-Expose-Headers header.
    exposedHeaders: false
    # Sets the Access-Control-Max-Age header.
    maxAge: false
    # Sets the Access-Control-Allow-Credentials header.
    supportsCredentials: false
```
- Create `.eslintignore` in the root of this project with `*` as its contents.
- Create `.env.local.js` from the example file, and update the hostname:
```
hostname: "d8.fs-demo.local",
```
- `yarn start`
- Test the search and facets to verify you see the expected results.
